tests/chainer_tests/test_function_node.py

```diff
@@ -426,14 +426,27 @@ def forward(self, inputs):
             f.apply((v,))
 
 
-class TestFunctionNodeInconsistentBackends(unittest.TestCase):
+class TestFunctionNodeForwardTypeCheck(unittest.TestCase):
 
     def setUp(self):
         self.x1 = numpy.random.rand(2, 3).astype(numpy.float32)
         self.x2 = numpy.random.rand(2, 3).astype(numpy.float32)
 
+    def test_invalid_output_type(self):
+        class FunctionNode(chainer.FunctionNode):
+
+            def forward(self, inputs):
+                return object(),
+
+        f = FunctionNode()
+        x1 = chainer.Variable(self.x1)
+
+        with self.assertRaisesRegex(
```

Comment:
Should use `six.assertRaisesRegex(self, ...)` for Py2 compatibility.

Reply:
Thanks, fixed as suggested.